homeassistant/components/modbus/sensor.py

```diff
@@ -121,6 +122,9 @@ async def async_setup_platform(
             del entry[CONF_REGISTER_TYPE]
 
     for entry in discovery_info[CONF_SENSORS]:
+
+        regSize = entry[CONF_COUNT] * 2 * entry.get(CONF_REGISTER_SIZE, 1)
```

Comment:
How is that going to work, the name "register_size" tells it is the size of a register (e.g. 2 or 4 bytes) you use it as a multiplicator. If it is a multiplicador the name need to reflect that.

If you say ALL devices have a multiple of 2 for which there are no guarantee a multiplicator would work, but it is not easy to understand. Why not keep register_size as the size of a register.